gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/src/main/java/io/gravitee/apim/gateway/tests/sdk/runner/GatewayRunner.java

```diff
@@ -331,11 +335,13 @@ public boolean isRunning() {
      * @param deployedApis the map containing deployed apis.
      */
     private void deploy(ReactableApi<?> reactableApi, Map<String, ReactableApi<?>> deployedApis) {
+        final GatewayMode mode = testInstance.getClass().getAnnotation(GatewayTest.class).mode();
```

Comment:
since you already do that previously, maybe it deserves to extract that has a field ?